crates/net/common/src/bandwidth_meter.rs

```diff
@@ -268,4 +344,74 @@ mod tests {
         assert_bandwidth_counts(&shared_client_bandwidth_meter, 8, 8);
         assert_bandwidth_counts(&shared_server_bandwidth_meter, 8, 8);
     }
+
+    struct ThroughputStatsTask<'a> {
+        interval: Interval,
+        meter: &'a BandwidthMeter,
+        max_throughputs: (f64, f64),
+    }
+
+    impl<'a> ThroughputStatsTask<'a> {
+        pub fn new(
+            interval: Interval,
+            meter: &'a BandwidthMeter,
+        ) -> Self {
+            Self {
+                interval,
+                meter,
+                max_throughputs: (0.0, 0.0)
+            }
+        }
+    }
+
+    impl<'a> Future for ThroughputStatsTask<'a> {
+        type Output = (f64, f64);
+
+        fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+            let this = self.get_mut();
+
+            let ingress_throughput = this.meter.ingress_throughput();
+            let egress_throughput = this.meter.egress_throughput();
+
+            this.max_throughputs = (
+                if ingress_throughput > this.max_throughputs.0 { ingress_throughput } else { this.max_throughputs.0 },
+                if egress_throughput > this.max_throughputs.1 { egress_throughput } else { this.max_throughputs.1 },
+            );
+
+            if this.interval.poll_tick(cx).is_ready() {
+                return Poll::Ready(this.max_throughputs)
+            }
+
+            Poll::Pending
+        }
+    }
+
+    #[tokio::test]
+    async fn test_throughput_basic() {
+        let (client, server) = duplex(64);
+        let (mut metered_client, mut _metered_server) =
+            (MeteredStream::new(client), MeteredStream::new(server));
+
+        let network_io_meter = metered_client.get_bandwidth_meter().clone();
+
+        // Concurrently collect throughput stats & write out from stream.
+        // Want to be sure that stats task starts first, but can't use `spawn` due to `'static` lifetime constraint...
+        let ((_, max_egress_throughput), _) = tokio::join!(
```

Comment:
Maybe not the cleanest code, but shows that I had trouble with the ergonomics of reading throughput off of the `BandwidthMeter`. It can be tricky to read it at exactly the right time, so I had to spawn a background task that periodically polls and keeps track of the max...

This pattern could be okay, but a couple potential alternatives (that I haven't thought too deeply about):
1. `ThroughputTask` can write the throughput values to a stream that can be consumed / polled on demand, instead of needing a background task that continually reads the values off of `BandwidthMeter`. The downside here is that if the stream is not being read somewhere else, we'll block on writing values into it, unless we use an unbounded channel.
2. We can make a top-down call about what the aggregate throughput measurement that we want is (e.g. moving average over the last 1min) and just calculate & store that metric directly on the `BandwidthMeter`. This is less flexible but means that we can just read throughput values directly off of the `BandwidthMeter` and know that we have the measurements we want, and not worry about having to collect & calculate out-of-band